[2022-03-28 05:27:03,116] [autoreload.py:637] [autoreload:run_with_reloader] [INFO]- Watching for file changes with StatReloader
[2022-03-28 05:33:19,442] [basehttp.py:161] [basehttp:log_message] [INFO]- "GET /media/web/img/class/3.jpg HTTP/1.1" 304 0
[2022-03-28 05:33:19,443] [basehttp.py:161] [basehttp:log_message] [INFO]- "GET /media/web/img/class/4.jpg HTTP/1.1" 304 0
[2022-03-28 05:33:19,443] [basehttp.py:161] [basehttp:log_message] [INFO]- "GET /media/web/img/class/1.jpg HTTP/1.1" 304 0
[2022-03-28 05:33:19,443] [basehttp.py:161] [basehttp:log_message] [INFO]- "GET /media/web/img/class/2.jpg HTTP/1.1" 304 0
[2022-03-28 05:33:19,561] [basehttp.py:161] [basehttp:log_message] [INFO]- "GET /api/class/get_recommend HTTP/1.1" 200 1353
[2022-03-28 05:33:19,751] [basehttp.py:161] [basehttp:log_message] [INFO]- "GET /media/web/img/class/652.jpg HTTP/1.1" 200 47786
[2022-03-28 05:33:19,753] [basehttp.py:161] [basehttp:log_message] [INFO]- "GET /media/web/img/class/default.jpg HTTP/1.1" 304 0
[2022-03-28 05:33:20,071] [basehttp.py:161] [basehttp:log_message] [INFO]- "GET /media/web/img/class/1069.jpg HTTP/1.1" 200 4032937
[2022-03-28 05:33:22,716] [basehttp.py:161] [basehttp:log_message] [INFO]- "GET /api/users/nick_repeat?nick=pxw%40163.co HTTP/1.1" 200 36
[2022-03-28 05:33:23,173] [basehttp.py:161] [basehttp:log_message] [INFO]- "GET /api/users/nick_repeat?nick=pxw%40163.co HTTP/1.1" 200 36
[2022-03-28 05:33:31,949] [basehttp.py:161] [basehttp:log_message] [INFO]- "POST /api/users/log_in HTTP/1.1" 200 417
[2022-03-28 05:33:31,983] [basehttp.py:161] [basehttp:log_message] [INFO]- "GET /media/web/img/avatar/default.jpg HTTP/1.1" 304 0
[2022-03-28 05:33:36,544] [basehttp.py:161] [basehttp:log_message] [INFO]- "GET /api/class/get_all_types HTTP/1.1" 200 1022
[2022-03-28 05:35:31,968] [basehttp.py:161] [basehttp:log_message] [INFO]- "GET /api/class/get_all_types HTTP/1.1" 200 1022
[2022-03-28 05:35:48,215] [basehttp.py:161] [basehttp:log_message] [INFO]- "GET /api/class/get_all_types HTTP/1.1" 200 1022
[2022-03-28 05:36:01,904] [basehttp.py:161] [basehttp:log_message] [INFO]- "GET /api/users/nick_repeat?nick=pxw%40163.co HTTP/1.1" 200 36
[2022-03-28 05:36:02,348] [basehttp.py:161] [basehttp:log_message] [INFO]- "GET /api/users/nick_repeat?nick=pxw%40163.co HTTP/1.1" 200 36
[2022-03-28 05:36:11,722] [basehttp.py:161] [basehttp:log_message] [INFO]- "POST /api/users/log_in HTTP/1.1" 200 415
[2022-03-28 05:50:10,977] [basehttp.py:161] [basehttp:log_message] [INFO]- "GET /api/class/get_all_types HTTP/1.1" 200 1022
[2022-03-28 05:50:42,879] [basehttp.py:161] [basehttp:log_message] [INFO]- "GET /api/class/get_all_types HTTP/1.1" 200 1022
[2022-03-28 05:50:44,011] [basehttp.py:161] [basehttp:log_message] [INFO]- "GET /api/users/nick_repeat?nick=pxw%40163.co HTTP/1.1" 200 36
[2022-03-28 05:50:44,465] [basehttp.py:161] [basehttp:log_message] [INFO]- "GET /api/users/nick_repeat?nick=pxw%40163.co HTTP/1.1" 200 36
[2022-03-28 05:50:54,665] [basehttp.py:161] [basehttp:log_message] [INFO]- "POST /api/users/log_in HTTP/1.1" 200 419
[2022-03-28 05:51:12,312] [log.py:224] [log:log_response] [WARNING]- Not Found: /api/class/create
[2022-03-28 05:51:12,313] [basehttp.py:161] [basehttp:log_message] [WARNING]- "POST /api/class/create HTTP/1.1" 404 3316
[2022-03-28 05:51:50,252] [autoreload.py:251] [autoreload:trigger_reload] [INFO]- /Users/linshengfeng/学习/大四上/毕业设计/mooc_recommender_system/mooc_back_end/web/class/urls.py changed, reloading.
[2022-03-28 05:51:50,753] [autoreload.py:637] [autoreload:run_with_reloader] [INFO]- Watching for file changes with StatReloader
[2022-03-28 05:51:58,705] [basehttp.py:161] [basehttp:log_message] [INFO]- "POST /api/class/create HTTP/1.1" 200 76
[2022-03-28 05:51:58,867] [basehttp.py:161] [basehttp:log_message] [INFO]- "GET /api/class/get_recommend HTTP/1.1" 200 1475
[2022-03-28 05:51:58,967] [basehttp.py:161] [basehttp:log_message] [INFO]- "GET /media/web/img/class/677.jpg HTTP/1.1" 200 83268
[2022-03-28 05:51:58,967] [basehttp.py:161] [basehttp:log_message] [INFO]- "GET /media/web/img/class/136.jpg HTTP/1.1" 200 56546
[2022-03-28 05:53:07,214] [autoreload.py:637] [autoreload:run_with_reloader] [INFO]- Watching for file changes with StatReloader
[2022-03-28 05:53:13,849] [basehttp.py:161] [basehttp:log_message] [INFO]- "GET /api/class/get_all_types HTTP/1.1" 200 1022
[2022-03-28 05:55:10,730] [basehttp.py:161] [basehttp:log_message] [INFO]- "GET /api/class/get_all_types HTTP/1.1" 200 1022
[2022-03-28 05:55:14,377] [basehttp.py:161] [basehttp:log_message] [INFO]- "GET /api/class/get_recommend HTTP/1.1" 200 1404
[2022-03-28 05:55:14,467] [basehttp.py:161] [basehttp:log_message] [INFO]- "GET /media/web/img/class/1291.jpg HTTP/1.1" 200 61616
[2022-03-28 05:55:16,679] [basehttp.py:161] [basehttp:log_message] [INFO]- "GET /api/class/get_all_types HTTP/1.1" 200 1022
[2022-03-28 05:55:29,137] [basehttp.py:161] [basehttp:log_message] [INFO]- "POST /api/class/create HTTP/1.1" 200 76
[2022-03-28 05:55:29,463] [basehttp.py:161] [basehttp:log_message] [INFO]- "GET /api/class/get_recommend HTTP/1.1" 200 1343
[2022-03-28 05:55:29,484] [basehttp.py:161] [basehttp:log_message] [INFO]- "GET /media/web/img/class/307.jpg HTTP/1.1" 200 233996
[2022-03-28 05:55:29,485] [basehttp.py:161] [basehttp:log_message] [INFO]- "GET /media/web/img/class/235.jpg HTTP/1.1" 200 517849
[2022-03-28 05:55:35,493] [basehttp.py:161] [basehttp:log_message] [INFO]- "GET /api/class/get_recommend HTTP/1.1" 200 1373
[2022-03-28 05:55:35,895] [basehttp.py:161] [basehttp:log_message] [INFO]- "GET /media/web/img/class/276.jpg HTTP/1.1" 200 24034
[2022-03-28 05:55:35,897] [basehttp.py:161] [basehttp:log_message] [INFO]- "GET /media/web/img/class/1000.jpg HTTP/1.1" 200 75734
[2022-03-28 05:55:35,906] [basehttp.py:161] [basehttp:log_message] [INFO]- "GET /media/web/img/class/291.jpg HTTP/1.1" 200 1407115
[2022-03-28 05:55:36,888] [basehttp.py:161] [basehttp:log_message] [INFO]- "GET /api/users/nick_repeat?nick=pxw%40163.co HTTP/1.1" 200 36
[2022-03-28 05:55:37,341] [basehttp.py:161] [basehttp:log_message] [INFO]- "GET /api/users/nick_repeat?nick=pxw%40163.co HTTP/1.1" 200 36
[2022-03-28 05:55:49,686] [basehttp.py:161] [basehttp:log_message] [INFO]- "POST /api/users/log_in HTTP/1.1" 200 419
[2022-03-28 05:55:55,409] [basehttp.py:161] [basehttp:log_message] [INFO]- "GET /api/class/get_all_types HTTP/1.1" 200 1022
[2022-03-28 05:56:51,533] [autoreload.py:637] [autoreload:run_with_reloader] [INFO]- Watching for file changes with StatReloader
[2022-03-28 05:58:49,062] [basehttp.py:161] [basehttp:log_message] [INFO]- "GET /api/class/get_recommend HTTP/1.1" 200 1375
[2022-03-28 05:58:49,128] [basehttp.py:161] [basehttp:log_message] [INFO]- "GET /media/web/img/class/604.jpg HTTP/1.1" 304 0
[2022-03-28 05:58:49,129] [basehttp.py:161] [basehttp:log_message] [INFO]- "GET /media/web/img/class/186.jpg HTTP/1.1" 304 0
[2022-03-28 05:58:49,130] [basehttp.py:161] [basehttp:log_message] [INFO]- "GET /media/web/img/class/225.jpg HTTP/1.1" 200 48879
[2022-03-28 05:58:49,132] [basehttp.py:161] [basehttp:log_message] [INFO]- "GET /media/web/img/class/760.jpg HTTP/1.1" 200 64923
[2022-03-28 05:58:50,952] [basehttp.py:161] [basehttp:log_message] [INFO]- "GET /api/class/get_recommend HTTP/1.1" 200 1380
[2022-03-28 05:58:51,346] [basehttp.py:161] [basehttp:log_message] [INFO]- "GET /media/web/img/class/436.jpg HTTP/1.1" 200 44411
[2022-03-28 05:58:51,348] [basehttp.py:161] [basehttp:log_message] [INFO]- "GET /media/web/img/class/707.jpg HTTP/1.1" 200 116128
[2022-03-28 05:58:51,348] [basehttp.py:161] [basehttp:log_message] [INFO]- "GET /media/web/img/class/498.jpg HTTP/1.1" 200 103175
[2022-03-28 05:58:51,349] [basehttp.py:161] [basehttp:log_message] [INFO]- "GET /media/web/img/class/239.jpg HTTP/1.1" 200 191255
[2022-03-28 05:58:52,052] [basehttp.py:161] [basehttp:log_message] [INFO]- "GET /api/users/nick_repeat?nick=pxw%40163.co HTTP/1.1" 200 36
[2022-03-28 05:58:52,512] [basehttp.py:161] [basehttp:log_message] [INFO]- "GET /api/users/nick_repeat?nick=pxw%40163.co HTTP/1.1" 200 36
[2022-03-28 05:59:01,026] [basehttp.py:161] [basehttp:log_message] [INFO]- "POST /api/users/log_in HTTP/1.1" 200 417
[2022-03-28 05:59:14,923] [basehttp.py:161] [basehttp:log_message] [INFO]- "GET /api/class/get_all_types HTTP/1.1" 200 1022
[2022-03-28 05:59:31,095] [basehttp.py:161] [basehttp:log_message] [INFO]- "POST /api/class/create HTTP/1.1" 200 77
[2022-03-28 05:59:31,157] [basehttp.py:161] [basehttp:log_message] [INFO]- "GET /api/class/get_recommend HTTP/1.1" 200 1312
[2022-03-28 05:59:31,188] [basehttp.py:161] [basehttp:log_message] [INFO]- "GET /media/web/img/class/981.jpg HTTP/1.1" 200 87616
[2022-03-28 06:00:12,591] [autoreload.py:251] [autoreload:trigger_reload] [INFO]- /Users/linshengfeng/学习/大四上/毕业设计/mooc_recommender_system/mooc_back_end/web/class/verify.py changed, reloading.
[2022-03-28 06:00:13,006] [autoreload.py:637] [autoreload:run_with_reloader] [INFO]- Watching for file changes with StatReloader
[2022-03-28 06:00:19,549] [autoreload.py:637] [autoreload:run_with_reloader] [INFO]- Watching for file changes with StatReloader
[2022-03-28 06:00:22,007] [basehttp.py:161] [basehttp:log_message] [INFO]- "GET /api/class/get_all_types HTTP/1.1" 200 1022
[2022-03-28 06:01:23,344] [middlewares.py:37] [middlewares:process_exception] [ERROR]- {'code': 5000, 'errmsg': '数据库错误', 'data': {}}
Traceback (most recent call last):
  File "/Users/linshengfeng/miniforge3/envs/py1.8/lib/python3.8/site-packages/django/db/backends/utils.py", line 84, in _execute
    return self.cursor.execute(sql, params)
  File "/Users/linshengfeng/miniforge3/envs/py1.8/lib/python3.8/site-packages/django/db/backends/mysql/base.py", line 73, in execute
    return self.cursor.execute(query, args)
  File "/Users/linshengfeng/miniforge3/envs/py1.8/lib/python3.8/site-packages/pymysql/cursors.py", line 148, in execute
    result = self._query(query)
  File "/Users/linshengfeng/miniforge3/envs/py1.8/lib/python3.8/site-packages/pymysql/cursors.py", line 310, in _query
    conn.query(q)
  File "/Users/linshengfeng/miniforge3/envs/py1.8/lib/python3.8/site-packages/pymysql/connections.py", line 548, in query
    self._affected_rows = self._read_query_result(unbuffered=unbuffered)
  File "/Users/linshengfeng/miniforge3/envs/py1.8/lib/python3.8/site-packages/pymysql/connections.py", line 775, in _read_query_result
    result.read()
  File "/Users/linshengfeng/miniforge3/envs/py1.8/lib/python3.8/site-packages/pymysql/connections.py", line 1156, in read
    first_packet = self.connection._read_packet()
  File "/Users/linshengfeng/miniforge3/envs/py1.8/lib/python3.8/site-packages/pymysql/connections.py", line 725, in _read_packet
    packet.raise_for_error()
  File "/Users/linshengfeng/miniforge3/envs/py1.8/lib/python3.8/site-packages/pymysql/protocol.py", line 221, in raise_for_error
    err.raise_mysql_exception(self._data)
  File "/Users/linshengfeng/miniforge3/envs/py1.8/lib/python3.8/site-packages/pymysql/err.py", line 143, in raise_mysql_exception
    raise errorclass(errno, errval)
pymysql.err.IntegrityError: (1048, "Column 'id' cannot be null")

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/linshengfeng/miniforge3/envs/py1.8/lib/python3.8/site-packages/django/core/handlers/base.py", line 181, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/Users/linshengfeng/miniforge3/envs/py1.8/lib/python3.8/site-packages/django/views/decorators/csrf.py", line 54, in wrapped_view
    return view_func(*args, **kwargs)
  File "/Users/linshengfeng/miniforge3/envs/py1.8/lib/python3.8/site-packages/django/views/decorators/http.py", line 40, in inner
    return func(request, *args, **kwargs)
  File "/Users/linshengfeng/学习/大四上/毕业设计/mooc_recommender_system/mooc_back_end/web/class/views.py", line 71, in createClass
    c.save()
  File "/Users/linshengfeng/miniforge3/envs/py1.8/lib/python3.8/site-packages/django/db/models/base.py", line 726, in save
    self.save_base(using=using, force_insert=force_insert,
  File "/Users/linshengfeng/miniforge3/envs/py1.8/lib/python3.8/site-packages/django/db/models/base.py", line 763, in save_base
    updated = self._save_table(
  File "/Users/linshengfeng/miniforge3/envs/py1.8/lib/python3.8/site-packages/django/db/models/base.py", line 868, in _save_table
    results = self._do_insert(cls._base_manager, using, fields, returning_fields, raw)
  File "/Users/linshengfeng/miniforge3/envs/py1.8/lib/python3.8/site-packages/django/db/models/base.py", line 906, in _do_insert
    return manager._insert(
  File "/Users/linshengfeng/miniforge3/envs/py1.8/lib/python3.8/site-packages/django/db/models/manager.py", line 85, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/Users/linshengfeng/miniforge3/envs/py1.8/lib/python3.8/site-packages/django/db/models/query.py", line 1270, in _insert
    return query.get_compiler(using=using).execute_sql(returning_fields)
  File "/Users/linshengfeng/miniforge3/envs/py1.8/lib/python3.8/site-packages/django/db/models/sql/compiler.py", line 1416, in execute_sql
    cursor.execute(sql, params)
  File "/Users/linshengfeng/miniforge3/envs/py1.8/lib/python3.8/site-packages/django/db/backends/utils.py", line 98, in execute
    return super().execute(sql, params)
  File "/Users/linshengfeng/miniforge3/envs/py1.8/lib/python3.8/site-packages/django/db/backends/utils.py", line 66, in execute
    return self._execute_with_wrappers(sql, params, many=False, executor=self._execute)
  File "/Users/linshengfeng/miniforge3/envs/py1.8/lib/python3.8/site-packages/django/db/backends/utils.py", line 75, in _execute_with_wrappers
    return executor(sql, params, many, context)
  File "/Users/linshengfeng/miniforge3/envs/py1.8/lib/python3.8/site-packages/django/db/backends/utils.py", line 84, in _execute
    return self.cursor.execute(sql, params)
  File "/Users/linshengfeng/miniforge3/envs/py1.8/lib/python3.8/site-packages/django/db/utils.py", line 90, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "/Users/linshengfeng/miniforge3/envs/py1.8/lib/python3.8/site-packages/django/db/backends/utils.py", line 84, in _execute
    return self.cursor.execute(sql, params)
  File "/Users/linshengfeng/miniforge3/envs/py1.8/lib/python3.8/site-packages/django/db/backends/mysql/base.py", line 73, in execute
    return self.cursor.execute(query, args)
  File "/Users/linshengfeng/miniforge3/envs/py1.8/lib/python3.8/site-packages/pymysql/cursors.py", line 148, in execute
    result = self._query(query)
  File "/Users/linshengfeng/miniforge3/envs/py1.8/lib/python3.8/site-packages/pymysql/cursors.py", line 310, in _query
    conn.query(q)
  File "/Users/linshengfeng/miniforge3/envs/py1.8/lib/python3.8/site-packages/pymysql/connections.py", line 548, in query
    self._affected_rows = self._read_query_result(unbuffered=unbuffered)
  File "/Users/linshengfeng/miniforge3/envs/py1.8/lib/python3.8/site-packages/pymysql/connections.py", line 775, in _read_query_result
    result.read()
  File "/Users/linshengfeng/miniforge3/envs/py1.8/lib/python3.8/site-packages/pymysql/connections.py", line 1156, in read
    first_packet = self.connection._read_packet()
  File "/Users/linshengfeng/miniforge3/envs/py1.8/lib/python3.8/site-packages/pymysql/connections.py", line 725, in _read_packet
    packet.raise_for_error()
  File "/Users/linshengfeng/miniforge3/envs/py1.8/lib/python3.8/site-packages/pymysql/protocol.py", line 221, in raise_for_error
    err.raise_mysql_exception(self._data)
  File "/Users/linshengfeng/miniforge3/envs/py1.8/lib/python3.8/site-packages/pymysql/err.py", line 143, in raise_mysql_exception
    raise errorclass(errno, errval)
django.db.utils.IntegrityError: (1048, "Column 'id' cannot be null")
[2022-03-28 06:01:25,197] [log.py:224] [log:log_response] [ERROR]- Internal Server Error: /api/class/create
[2022-03-28 06:01:25,202] [basehttp.py:161] [basehttp:log_message] [ERROR]- "POST /api/class/create HTTP/1.1" 500 15
[2022-03-28 06:07:18,485] [autoreload.py:251] [autoreload:trigger_reload] [INFO]- /Users/linshengfeng/学习/大四上/毕业设计/mooc_recommender_system/mooc_back_end/web/class/views.py changed, reloading.
[2022-03-28 06:07:19,521] [autoreload.py:637] [autoreload:run_with_reloader] [INFO]- Watching for file changes with StatReloader
[2022-03-28 06:07:56,060] [autoreload.py:251] [autoreload:trigger_reload] [INFO]- /Users/linshengfeng/学习/大四上/毕业设计/mooc_recommender_system/mooc_back_end/web/class/views.py changed, reloading.
[2022-03-28 06:07:56,963] [autoreload.py:637] [autoreload:run_with_reloader] [INFO]- Watching for file changes with StatReloader
[2022-03-28 06:12:01,508] [autoreload.py:251] [autoreload:trigger_reload] [INFO]- /Users/linshengfeng/学习/大四上/毕业设计/mooc_recommender_system/mooc_back_end/web/class/views.py changed, reloading.
[2022-03-28 06:12:02,487] [autoreload.py:637] [autoreload:run_with_reloader] [INFO]- Watching for file changes with StatReloader
[2022-03-28 06:12:42,606] [autoreload.py:251] [autoreload:trigger_reload] [INFO]- /Users/linshengfeng/学习/大四上/毕业设计/mooc_recommender_system/mooc_back_end/web/class/views.py changed, reloading.
[2022-03-28 06:12:43,581] [autoreload.py:637] [autoreload:run_with_reloader] [INFO]- Watching for file changes with StatReloader
[2022-03-28 06:13:03,049] [autoreload.py:251] [autoreload:trigger_reload] [INFO]- /Users/linshengfeng/学习/大四上/毕业设计/mooc_recommender_system/mooc_back_end/web/class/views.py changed, reloading.
[2022-03-28 06:13:03,935] [autoreload.py:637] [autoreload:run_with_reloader] [INFO]- Watching for file changes with StatReloader
[2022-03-28 06:13:17,066] [autoreload.py:251] [autoreload:trigger_reload] [INFO]- /Users/linshengfeng/学习/大四上/毕业设计/mooc_recommender_system/mooc_back_end/web/class/views.py changed, reloading.
[2022-03-28 06:13:18,299] [autoreload.py:637] [autoreload:run_with_reloader] [INFO]- Watching for file changes with StatReloader
[2022-03-28 06:14:26,575] [autoreload.py:251] [autoreload:trigger_reload] [INFO]- /Users/linshengfeng/学习/大四上/毕业设计/mooc_recommender_system/mooc_back_end/web/class/views.py changed, reloading.
[2022-03-28 06:14:27,622] [autoreload.py:637] [autoreload:run_with_reloader] [INFO]- Watching for file changes with StatReloader
[2022-03-28 06:14:52,974] [autoreload.py:637] [autoreload:run_with_reloader] [INFO]- Watching for file changes with StatReloader
[2022-03-28 06:15:36,857] [autoreload.py:637] [autoreload:run_with_reloader] [INFO]- Watching for file changes with StatReloader
[2022-03-28 06:16:19,728] [autoreload.py:637] [autoreload:run_with_reloader] [INFO]- Watching for file changes with StatReloader
[2022-03-28 06:16:28,882] [autoreload.py:637] [autoreload:run_with_reloader] [INFO]- Watching for file changes with StatReloader
[2022-03-28 06:18:03,537] [basehttp.py:161] [basehttp:log_message] [INFO]- "POST /api/class/create HTTP/1.1" 200 70
[2022-03-28 06:18:03,774] [basehttp.py:161] [basehttp:log_message] [INFO]- "GET /api/class/get_recommend HTTP/1.1" 200 1363
[2022-03-28 06:18:03,837] [basehttp.py:161] [basehttp:log_message] [INFO]- "GET /media/web/img/class/203.jpg HTTP/1.1" 200 50238
[2022-03-28 06:27:04,112] [basehttp.py:161] [basehttp:log_message] [INFO]- "GET /api/class/get_all_types HTTP/1.1" 200 1022
[2022-03-28 06:28:22,447] [autoreload.py:637] [autoreload:run_with_reloader] [INFO]- Watching for file changes with StatReloader
[2022-03-28 06:29:31,571] [autoreload.py:637] [autoreload:run_with_reloader] [INFO]- Watching for file changes with StatReloader
[2022-03-28 06:29:45,255] [autoreload.py:637] [autoreload:run_with_reloader] [INFO]- Watching for file changes with StatReloader
[2022-03-28 06:38:07,676] [autoreload.py:637] [autoreload:run_with_reloader] [INFO]- Watching for file changes with StatReloader
[2022-03-28 06:38:09,264] [basehttp.py:161] [basehttp:log_message] [INFO]- "POST /api/users/log_in HTTP/1.1" 200 417
[2022-03-28 06:38:46,851] [basehttp.py:161] [basehttp:log_message] [INFO]- "GET /api/class/get_recommend HTTP/1.1" 200 1334
[2022-03-28 06:38:46,904] [basehttp.py:161] [basehttp:log_message] [INFO]- "GET /media/web/img/class/114.jpg HTTP/1.1" 304 0
[2022-03-28 06:38:48,997] [basehttp.py:161] [basehttp:log_message] [INFO]- "GET /api/class/get_all_types HTTP/1.1" 200 1022
[2022-03-28 06:39:31,963] [middlewares.py:43] [middlewares:process_exception] [ERROR]- {'code': 500, 'errmsg': '服务器异常', 'data': {}}
Traceback (most recent call last):
  File "/Users/linshengfeng/miniforge3/envs/py1.8/lib/python3.8/site-packages/django/core/handlers/base.py", line 181, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/Users/linshengfeng/miniforge3/envs/py1.8/lib/python3.8/site-packages/django/views/decorators/csrf.py", line 54, in wrapped_view
    return view_func(*args, **kwargs)
  File "/Users/linshengfeng/miniforge3/envs/py1.8/lib/python3.8/site-packages/django/views/decorators/http.py", line 40, in inner
    return func(request, *args, **kwargs)
  File "/Users/linshengfeng/学习/大四上/毕业设计/mooc_recommender_system/mooc_back_end/web/class/views.py", line 70, in createClass
    class_data, request = verify_class_params(request)
  File "/Users/linshengfeng/学习/大四上/毕业设计/mooc_recommender_system/mooc_back_end/web/class/verify.py", line 22, in verify_class_params
    class_data = json.loads(request.body.decode('utf-8'))
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xff in position 146: invalid start byte
[2022-03-28 06:39:31,969] [log.py:224] [log:log_response] [ERROR]- Internal Server Error: /api/class/create
[2022-03-28 06:39:31,970] [basehttp.py:161] [basehttp:log_message] [ERROR]- "POST /api/class/create HTTP/1.1" 500 15
[2022-03-28 06:40:15,608] [autoreload.py:251] [autoreload:trigger_reload] [INFO]- /Users/linshengfeng/学习/大四上/毕业设计/mooc_recommender_system/mooc_back_end/web/class/verify.py changed, reloading.
[2022-03-28 06:40:16,649] [autoreload.py:637] [autoreload:run_with_reloader] [INFO]- Watching for file changes with StatReloader
[2022-03-28 06:40:20,664] [middlewares.py:43] [middlewares:process_exception] [ERROR]- {'code': 500, 'errmsg': '服务器异常', 'data': {}}
Traceback (most recent call last):
  File "/Users/linshengfeng/miniforge3/envs/py1.8/lib/python3.8/site-packages/django/core/handlers/base.py", line 181, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/Users/linshengfeng/miniforge3/envs/py1.8/lib/python3.8/site-packages/django/views/decorators/csrf.py", line 54, in wrapped_view
    return view_func(*args, **kwargs)
  File "/Users/linshengfeng/miniforge3/envs/py1.8/lib/python3.8/site-packages/django/views/decorators/http.py", line 40, in inner
    return func(request, *args, **kwargs)
  File "/Users/linshengfeng/学习/大四上/毕业设计/mooc_recommender_system/mooc_back_end/web/class/views.py", line 70, in createClass
    class_data, request = verify_class_params(request)
  File "/Users/linshengfeng/学习/大四上/毕业设计/mooc_recommender_system/mooc_back_end/web/class/verify.py", line 23, in verify_class_params
    c_name = class_data.get('name', None)
AttributeError: 'WSGIRequest' object has no attribute 'get'
[2022-03-28 06:40:20,671] [log.py:224] [log:log_response] [ERROR]- Internal Server Error: /api/class/create
[2022-03-28 06:40:20,672] [basehttp.py:161] [basehttp:log_message] [ERROR]- "POST /api/class/create HTTP/1.1" 500 15
[2022-03-28 06:41:25,486] [autoreload.py:637] [autoreload:run_with_reloader] [INFO]- Watching for file changes with StatReloader
[2022-03-28 06:41:42,049] [middlewares.py:43] [middlewares:process_exception] [ERROR]- {'code': 500, 'errmsg': '服务器异常', 'data': {}}
Traceback (most recent call last):
  File "/Users/linshengfeng/miniforge3/envs/py1.8/lib/python3.8/site-packages/django/core/handlers/base.py", line 181, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/Users/linshengfeng/miniforge3/envs/py1.8/lib/python3.8/site-packages/django/views/decorators/csrf.py", line 54, in wrapped_view
    return view_func(*args, **kwargs)
  File "/Users/linshengfeng/miniforge3/envs/py1.8/lib/python3.8/site-packages/django/views/decorators/http.py", line 40, in inner
    return func(request, *args, **kwargs)
  File "/Users/linshengfeng/学习/大四上/毕业设计/mooc_recommender_system/mooc_back_end/web/class/views.py", line 70, in createClass
    class_data, request = verify_class_params(request)
  File "/Users/linshengfeng/学习/大四上/毕业设计/mooc_recommender_system/mooc_back_end/web/class/verify.py", line 22, in verify_class_params
    class_data = request['POST']
TypeError: 'WSGIRequest' object is not subscriptable
[2022-03-28 06:41:42,293] [log.py:224] [log:log_response] [ERROR]- Internal Server Error: /api/class/create
[2022-03-28 06:41:42,297] [basehttp.py:161] [basehttp:log_message] [ERROR]- "POST /api/class/create HTTP/1.1" 500 15
[2022-03-28 06:42:09,700] [autoreload.py:637] [autoreload:run_with_reloader] [INFO]- Watching for file changes with StatReloader
[2022-03-28 06:42:39,058] [middlewares.py:43] [middlewares:process_exception] [ERROR]- {'code': 500, 'errmsg': '服务器异常', 'data': {}}
Traceback (most recent call last):
  File "/Users/linshengfeng/miniforge3/envs/py1.8/lib/python3.8/site-packages/django/core/handlers/base.py", line 181, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/Users/linshengfeng/miniforge3/envs/py1.8/lib/python3.8/site-packages/django/views/decorators/csrf.py", line 54, in wrapped_view
    return view_func(*args, **kwargs)
  File "/Users/linshengfeng/miniforge3/envs/py1.8/lib/python3.8/site-packages/django/views/decorators/http.py", line 40, in inner
    return func(request, *args, **kwargs)
  File "/Users/linshengfeng/学习/大四上/毕业设计/mooc_recommender_system/mooc_back_end/web/class/views.py", line 70, in createClass
    class_data, request = verify_class_params(request)
  File "/Users/linshengfeng/学习/大四上/毕业设计/mooc_recommender_system/mooc_back_end/web/class/verify.py", line 31, in verify_class_params
    if not 0 <= c_type <= 23:
TypeError: '<=' not supported between instances of 'int' and 'str'
[2022-03-28 06:42:39,789] [log.py:224] [log:log_response] [ERROR]- Internal Server Error: /api/class/create
[2022-03-28 06:42:39,794] [basehttp.py:161] [basehttp:log_message] [ERROR]- "POST /api/class/create HTTP/1.1" 500 15
[2022-03-28 06:43:44,678] [autoreload.py:637] [autoreload:run_with_reloader] [INFO]- Watching for file changes with StatReloader
[2022-03-28 06:44:16,511] [basehttp.py:161] [basehttp:log_message] [INFO]- "POST /api/class/create HTTP/1.1" 200 70
[2022-03-28 06:44:16,712] [basehttp.py:161] [basehttp:log_message] [INFO]- "GET /api/class/get_recommend HTTP/1.1" 200 1290
[2022-03-28 06:44:16,873] [basehttp.py:161] [basehttp:log_message] [INFO]- "GET /media/web/img/class/990.jpg HTTP/1.1" 200 13148
[2022-03-28 06:44:16,875] [basehttp.py:161] [basehttp:log_message] [INFO]- "GET /media/web/img/class/621.jpg HTTP/1.1" 200 42029
[2022-03-28 06:44:16,877] [basehttp.py:161] [basehttp:log_message] [INFO]- "GET /media/web/img/class/340.jpg HTTP/1.1" 200 158799
[2022-03-28 06:46:11,018] [autoreload.py:251] [autoreload:trigger_reload] [INFO]- /Users/linshengfeng/学习/大四上/毕业设计/mooc_recommender_system/mooc_back_end/web/class/util.py changed, reloading.
[2022-03-28 06:46:12,012] [autoreload.py:637] [autoreload:run_with_reloader] [INFO]- Watching for file changes with StatReloader
[2022-03-28 06:47:06,395] [autoreload.py:637] [autoreload:run_with_reloader] [INFO]- Watching for file changes with StatReloader
[2022-03-28 06:47:12,391] [basehttp.py:161] [basehttp:log_message] [INFO]- "GET /api/class/get_recommend HTTP/1.1" 200 1460
[2022-03-28 06:47:12,828] [basehttp.py:161] [basehttp:log_message] [INFO]- "GET /media/web/img/class/633.jpg HTTP/1.1" 304 0
[2022-03-28 06:47:12,830] [basehttp.py:161] [basehttp:log_message] [INFO]- "GET /media/web/img/class/343.jpg HTTP/1.1" 304 0
[2022-03-28 06:47:12,831] [basehttp.py:161] [basehttp:log_message] [INFO]- "GET /media/web/img/class/76.jpg HTTP/1.1" 304 0
[2022-03-28 06:47:12,831] [basehttp.py:161] [basehttp:log_message] [INFO]- "GET /media/web/img/class/1304.jpg HTTP/1.1" 200 60547
[2022-03-28 06:47:34,886] [basehttp.py:161] [basehttp:log_message] [INFO]- "GET /api/class/get_all_types HTTP/1.1" 200 1022
[2022-03-28 06:47:34,896] [log.py:224] [log:log_response] [WARNING]- Not Found: /api/class/get_all_recommends/undefined/0/0
[2022-03-28 06:47:34,897] [basehttp.py:161] [basehttp:log_message] [WARNING]- "GET /api/class/get_all_recommends/undefined/0/0 HTTP/1.1" 404 3603
[2022-03-28 06:47:40,692] [basehttp.py:161] [basehttp:log_message] [INFO]- "GET /api/class/get_recommend HTTP/1.1" 200 1381
[2022-03-28 06:47:52,765] [basehttp.py:161] [basehttp:log_message] [INFO]- "GET /api/users/nick_repeat?nick=pxw%40163.co HTTP/1.1" 200 36
[2022-03-28 06:47:53,221] [basehttp.py:161] [basehttp:log_message] [INFO]- "GET /api/users/nick_repeat?nick=pxw%40163.co HTTP/1.1" 200 36
[2022-03-28 06:47:55,355] [basehttp.py:161] [basehttp:log_message] [INFO]- "POST /api/users/log_in HTTP/1.1" 200 459
[2022-03-28 06:47:56,908] [basehttp.py:161] [basehttp:log_message] [INFO]- "GET /api/class/get_all_types HTTP/1.1" 200 1022
[2022-03-28 06:47:59,283] [basehttp.py:161] [basehttp:log_message] [INFO]- "GET /api/class/get_all_recommends/7/0/0 HTTP/1.1" 200 5396
[2022-03-28 06:47:59,305] [basehttp.py:161] [basehttp:log_message] [INFO]- "GET /media/web/img/class/9.jpg HTTP/1.1" 304 0
[2022-03-28 06:47:59,305] [basehttp.py:161] [basehttp:log_message] [INFO]- "GET /media/web/img/class/10.jpg HTTP/1.1" 304 0
[2022-03-28 06:47:59,306] [basehttp.py:161] [basehttp:log_message] [INFO]- "GET /media/web/img/class/6.jpg HTTP/1.1" 304 0
[2022-03-28 06:47:59,308] [basehttp.py:161] [basehttp:log_message] [INFO]- "GET /media/web/img/class/14.jpg HTTP/1.1" 304 0
[2022-03-28 06:47:59,310] [basehttp.py:161] [basehttp:log_message] [INFO]- "GET /media/web/img/class/17.jpg HTTP/1.1" 304 0
[2022-03-28 06:47:59,310] [basehttp.py:161] [basehttp:log_message] [INFO]- "GET /media/web/img/class/19.jpg HTTP/1.1" 304 0
[2022-03-28 06:47:59,310] [basehttp.py:161] [basehttp:log_message] [INFO]- "GET /media/web/img/class/18.jpg HTTP/1.1" 304 0
[2022-03-28 06:48:26,717] [basehttp.py:161] [basehttp:log_message] [INFO]- "GET /api/class/get_all_recommends/7/1300/0 HTTP/1.1" 200 1342
[2022-03-28 06:48:26,732] [basehttp.py:161] [basehttp:log_message] [INFO]- "GET /media/web/img/class/1301.jpg HTTP/1.1" 304 0
[2022-03-28 06:48:41,322] [basehttp.py:161] [basehttp:log_message] [INFO]- "GET /api/class/get_all_recommends/7/0/0 HTTP/1.1" 200 5396
[2022-03-28 06:48:52,482] [basehttp.py:161] [basehttp:log_message] [INFO]- "GET /api/class/get_recommend HTTP/1.1" 200 1430
